packages/models-library/src/models_library/project_nodes.py

```diff
@@ -0,0 +1,188 @@
+from enum import Enum
+from typing import Dict, List, Optional, Union
+from uuid import UUID
+
+from pydantic import AnyUrl, BaseModel, Extra, Field, HttpUrl, constr, validator
+
+from .services import KEY_RE, PROPERTY_KEY_RE, VERSION_RE
+
+NodeID = UUID
+
+
+class RunningState(str, Enum):
+    UNKNOWN = "UNKNOWN"
+    PUBLISHED = "PUBLISHED"
+    NOT_STARTED = "NOT_STARTED"
+    PENDING = "PENDING"
+    STARTED = "STARTED"
+    RETRY = "RETRY"
+    SUCCESS = "SUCCESS"
+    FAILED = "FAILED"
+    ABORTED = "ABORTED"
+
+
+class Position(BaseModel):
+    x: int = Field(..., description="The x position", example=["12"])
+    y: int = Field(..., description="The y position", example=["15"])
+
+    class Config:
+        extra = Extra.forbid
+
+
+class AccessEnum(str, Enum):
+    ReadAndWrite = "ReadAndWrite"
+    Invisible = "Invisible"
+    ReadOnly = "ReadOnly"
+
+
+class PortLink(BaseModel):
+    nodeUuid: NodeID = Field(
+        ...,
+        description="The node to get the port output from",
+        example=["da5068e0-8a8d-4fb9-9516-56e5ddaef15b"],
+    )
+    output: str = Field(
+        ...,
+        description="The port key in the node given by nodeUuid",
+        regex=PROPERTY_KEY_RE,
+        example=["out_2"],
+    )
+
+    class Config:
+        extra = Extra.forbid
+
+
+class DownloadLink(BaseModel):
+    download_link: AnyUrl = Field(...)
+    label: Optional[str]
+
+    class Config:
+        extra = Extra.forbid
+
+
+class BaseFileLink(BaseModel):
+    store: Union[str, int] = Field(
```

Comment:
MINOR: do we really need both ``str`` and ``int``? pydantic is able to parse & convert from ``str`` to ``int``. Again, this is not relevant for this PR but we could do a PR with model only changes and refine these

Reply:
no it's not but it's like this at the moment for historical reasons